includes/multi-currency/Compatibility/WooCommerceSubscriptions.php

```diff
@@ -143,56 +145,36 @@ public function maybe_disable_mixed_cart( $value ) {
 	/**
 	 * Checks to see if the if the selected currency needs to be overridden.
 	 *
+	 * The running_override_selected_currency_filters property is used here to avoid infinite loops.
+	 *
 	 * @param mixed $return Default is false, but could be three letter currency code.
 	 *
 	 * @return mixed Three letter currency code or false if not.
 	 */
 	public function override_selected_currency( $return ) {
-		// If it's not false, return it.
+		// If it's not false, or we are already running filters, exit.
 		if ( $return || $this->running_override_selected_currency_filters ) {
 			return $return;
 		}
 
-		$subscription_renewal = $this->cart_contains_renewal();
-		if ( $subscription_renewal ) {
-			$order = wc_get_order( $subscription_renewal['subscription_renewal']['renewal_order_id'] );
-			return $order ? $order->get_currency() : $return;
-		}
+		// Loop through subscription types and check for cart items.
+		foreach ( self::SUBSCRIPTION_TYPES as $type ) {
```

Comment:
This loop now handles checking for a renewal, resubscribe, or switch in the cart. This includes `order_awaiting_payment`, which there was previously a check for.

Reply:
It looks like previously the `order_awaiting_payment` check would then check if that order contained a renewal. Is it correct for me to assume that the three types in the `SUBSCRIPTION_TYPES` would return true if checked for `( isset( WC()->session ) && WC()->session->get( 'order_awaiting_payment' ) )`? 

I just want to make sure I correctly understand how this loop includes an `order_awaiting_payment` check similar to the previous code.

Reply:
Yes, this helps. Thanks for clarifying!